### PR TITLE
feat(core): add wechat-web connector

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add wechat web connector and corresponding UTs.

## Linear Issue Reference

LOG-1895

## Testing

Check with a real WeChat web application:
image
image
image
image
image

All other UTs passed.

### CODE REVIEW

File: packages/core/src/connectors/types.ts [hunk 1]
```diff
@@ -87,9 +87,12 @@ export type ValidateConfig<T extends ArbitraryObject = ArbitraryObject> = (
...
 export type GetAuthorizationUri = (redirectUri: string, state: string) => Promise<string>;
 
-export type GetAccessToken = (code: string, redirectUri: string) => Promise<string>;
+export type GetAccessToken = (
+  code: string,
+  redirectUri?: string
+) => Promise<{ accessToken: string; openid?: string }>;
 
-export type GetUserInfo = (accessToken: string) => Promise<SocialUserInfo>;
+export type GetUserInfo = (accessToken: string, openid?: string) => Promise<SocialUserInfo>;
```
Comment (wangsijie) [inline packages/core/src/connectors/types.ts:95]: `userId`?

**Comment by IceHe** [comment]
Add some WeChat doc links in code comments?

File: packages/core/src/connectors/wechat/index.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+// eslint-disable-next-line unicorn/prefer-module
+const pathToReadmeFile = path.join(__dirname, 'README.md');
+const readmeContentFallback = 'Please check README.md file directory.';
+export const metadata: ConnectorMetadata = {
+  id: 'wechat',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with WeChat',
+    'zh-CN': 'WeChat 登录',
```
Comment (wangsijie) [inline packages/core/src/connectors/wechat/index.ts:38]: 微信登录？

File: packages/core/src/connectors/wechat/index.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    const userId = unionid ?? openid;
+
+    if (!userId) {
+      throw new ConnectorError(ConnectorErrorCodes.General);
+    }
+
+    return { id: userId, avatar: headimgurl, name: nickname };
+  } catch (error: unknown) {
+    if (error instanceof GotRequestError && error.response?.statusCode === 401) {
+      throw new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid);
```
Comment (wangsijie) [inline packages/core/src/connectors/wechat/index.ts:127]: Just a reminder: make sure to test this case in real env.

**Review by wangsijie — APPROVED** [review]

**Review by xiaoyijun — APPROVED** [review]
LGTM

File: packages/core/src/connectors/types.ts (hunk 1 above)
Comment (darcyYe) [reply packages/core/src/connectors/types.ts]: Fixed.

**Comment by darcyYe** [comment]
> Add some WeChat doc links in code comments?

The link is attached at the beginning of the wechat/index.ts file.

File: packages/core/src/connectors/wechat/index.ts (hunk 2 above)
Comment (darcyYe) [reply packages/core/src/connectors/wechat/index.ts]: Updated.

File: packages/core/src/connectors/wechat/index.ts (hunk 3 above)
Comment (darcyYe) [reply packages/core/src/connectors/wechat/index.ts:133]: Thanks for the mention :P

**Review by wangsijie — APPROVED** [review]

**Pull request merged**